torch/_decomp/decompositions.py

```diff
@@ -535,7 +535,7 @@ def binary_cross_entropy(
     #     "all elements of input should be between 0 and 1"
     # )
     loss = (target - 1) * torch.maximum(
-        torch.log(1 - self), self.new_full((), -100)
+        torch.log1p(-self), self.new_full((), -100)
     ) - target * torch.maximum(torch.log(self), self.new_full((), -100))
```

Comment:
If it weren't for the `torch.maximum`s, this could be golfed as
```python
target * x.logit() + x.neg().log1p()
```

Reply:
I'd assume the clamping is important, but I don't know why the author preferred `torch.maximum` over the cheaper `self.log().clamp(min=-100)`

Reply:
Yeah replacing maximums with clamp_mins would avoid those `new_fulls`, but I think this is getting optimized away anyway.